README.txt

"""
Copyright (C) 2013, 2014, 2015, 2016 Digital Freedom Foundation
  This file is part of GNUKhata:A modular,robust and Free Accounting System.

  GNUKhata is Free Software; you can redistribute it and/or modify
  it under the terms of the GNU Affero General Public License as
  published by the Free Software Foundation; either version 3 of
  the License, or (at your option) any later version.and old.stockflag = 's'

  GNUKhata is distributed in the hope that it will be useful, but
  WITHOUT ANY WARRANTY; without even the implied warranty of
  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  GNU Affero General Public License for more details.

  You should have received a copy of the GNU Affero General Public
  License along with GNUKhata (COPYING); if not, write to the
  Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
  Boston, MA  02110-1301  USA59 Temple Place, Suite 330,


Contributors:
"Krishnakant Mane" <kk@dff.org.in>
"Arun Kelkar" <arunkelkar@dff.org.in>
"Ishan Masdekar " <imasdekar@dff.org.in>
"Navin Karkera" <navin@dff.org.in>
"""

This is the web application frontend for GNUKhata.
This software works inside a browser and is known to work well with Firefox and Chromium.
Copyright (C) 2015 2016 Digital Freedom Foundation <www.dff.org.in>
GNUKhata is free software.
You can copy, Study, Modify and redistribute copys, either original or modified under the terms of GNU General Public License version 3 or at your choice any later version.


CHANGELOG
[19 June 2020 Release 7.0]
Upgraded the entire code base to Python 3 (Python3.8).
New searchable select box.
Narration box in Delivery note and Rejection note documents.
Documents details in ledger.
Modified print of invoices.
Total amount in words in deliverynote.
Security enhancements.
Performance enhancements.
Minor enhancements and bug fixes.


[16 December 2019 Release 6.50]
* More screens are responsive.
* Resolved bug that prevented Company Logo from being displayed.
* New format for printing invoice for retailers.
* Discount can be set in percent or as amount while creating a Product or Service. This gets loaded automatically while creating documents.
* Facility to cancel invoices associated with Delivery Notes.
* Facility to cancel Delivery Notes and a report listing all cancelled deliveries.
* Better search combo in select company screen and report screens
* Consistency in creating customer/supplier ledger account
* Auto populate Issuer name and designation field in documents
* Enhancement in rejection note.
* Sample charts and diagrams in dashboard when there is no data.
* Enhancement in Activity log.
* Minor enhancements and bug fixes


[16 August 2019 Release 6.0 Update]
* Payment button uncleared invoices.
* P&L and Balance Sheets available for any period within a financial year.
* Improvements in responsiveness.
* Facility to Create Invoice from Delivery Note.
* Facility to import bulk data in the form of spreadsheets on modules Account, Customer & Supplier, Product and Godown to ease data entry.
* Notes can be added to documents like Invoice, Debit & Credit Note.
* Round Off feature for Delivery note , Credit/Debit note and Purchase/sales order , Cashmemo with automatic accounting entries
* Cash/Bank balances visible in dashboard.
* Performance enhancements in dashboard.
* Checkbox to indicate if goods returned are of bad quality in Credit Note.
* Improvement in print templates for invoice.
* Any voucher can be deleted. Amount Paid for invoice gets adjusted automatically.
* PIN code can be added for Customer/Supplier.
* Both customers and suppliers are listed in create screens of documents.
* Security enhancements like setting Content Security Policy and response headers like x-frame-options.
* Improved navigation and shortcuts user and company preference wise.
* Performance enhancements.
* Bug fixes.

[26 April 2019 Release 6.0]
* Facility to create,edit and view report of Profit & Loss Budget and cash Budget.
* Added dynamic dashboard.
* Facility to cancel invoices and report listing all cancelled invoices.
* Details of transactions made can be viewed from the preview of invoice.
* Menu has been shifted from navbar to a new sidebar.
* Facility to Round off total value of invoice.
* Deprecated Accounting with Invoicing option.
* Removed themes that are not compatible.
* Searchable dropdown for list of accounts in edit account and view ledger screens.
* Added a new page for listing all reports.
* Spreadsheets are no longer saved as files but as string buffers. Now they can be download from Windows machines also.
* Minor bugs resolved. 
* More obvious way to add customer or supplier from documents viz.Invoice, delivery note and purchase/sales order.
* Prevented click jacking attack.
* A bank account is created when bank details are entered in company setup/edit organisation.
* Added '+' button in product table of transfer note.
* Some screens have been made more responsive.
* New tab for list of accounts in accounts module.
* Enhanced existing help screen added new ones.

[29 November 2018 Release 5.50]
* Implemented GSTR1 Report and Spreadsheet.  
* Presentation of Profit and Loss Report revised.  
* New fields in Product module to save MRP and Selling Price.  
* Last Price/MRP/Selling Price autoloads in Invoice.  
* Enhanced searching experience in dropdowns.  
* Plus button in product details table in documents such as invoice, delivery note etc  to add new row.  
* Facility to add product from Invoice.  
* Functionality to delete invoices with no transactions/debit note/credit note/rejection note.   
* Duplicate and Triplicate Printable versions of Invoice.  
* More formats for printing invoices.  
* Notes field in print invoice.  
* List of invoices in Invoice Module.  
* Security enhancements.  
* Drill down facility in unbilled deliveries.  
* Preview of delivery note and attatchment in view delivery note.   
* Minor bugs resolved.  

[15 August 2018 Release 5.25]
* Invoice attachments are getting saved again. 
* Narration of previous voucher can be used while creating a new voucher. 
* Fixed bugs related to Debit/Credit Note, Company Setup and Edit Profile. 
* Values in spreadsheet are now in float. 

[15 June 2018 Release 5.10]
* Debit Credit note for adjustment in quantity
* New company setup	
* Automated sales purchase entries with invoices
* Automated Debit Credit entry for Debit credit note
* Alternative sales purchase vouchers entries
* Rectify invoice with or without auto voucher
* Voucher entry without voucher no. and automatic voucher no. entry
* Narration repeatation	
* Calculator in voucher
* Sorting functionality in major modules
* Set/Reset default accounts for cash & bank & sales & purchase accounts.
* Chart of accounts
* Automated GST summary report
* System generated accounts for GST & Cess	
* Added more context sensitive help.	
* UI & UX enhancement done.	
* Fixed minor bugs.


[12 April 2018 Release 5.0]
* New menu "Documents".	
* Godown keeper can access list of all products and can add opening stock.
* Importing data from other software made more easy.
* Bankdetails in edit orgnisation page for organisation.
* GST compliant delivery chalan.
* GST compliant Purchase and sales order.
* GST compliant Rejection note.	
* GST compliant Debit and credit note.
* GST compatible register report. 	
* GST summary report.
* Rules corrected in invoice and cash memo to comply with GST.
* View of record cash memo, invoice after saving. 		

gkwebapp/__init__.py
"""
Copyright (C) 2013, 2014, 2015, 2016 Digital Freedom Foundation
Copyright (C) 2017, 2018 Digital Freedom Foundation & Accion Labs Pvt. Ltd.

  This file is part of GNUKhata:A modular,robust and Free Accounting System.

  GNUKhata is Free Software; you can redistribute it and/or modify
  it under the terms of the GNU Affero General Public License as
  published by the Free Software Foundation; either version 3 of
  the License, or (at your option) any later version.

  GNUKhata is distributed in the hope that it will be useful, but
  WITHOUT ANY WARRANTY; without even the implied warranty of
  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  GNU Affero General Public License for more details.

  You should have received a copy of the GNU Affero General Public
  License along with GNUKhata (COPYING); if not, write to the
  Free Software Foundation, Inc.,51 Franklin Street, 
  Fifth Floor, Boston, MA 02110, United States


Contributors:
"Krishnakant Mane" <kk@dff.org.in>
"Arun Kelkar" <arunkelkar@dff.org.in>
"Ishan Masdekar " <imasdekar@dff.org.in>
"Navin Karkera" <navin@dff.org.in>
"Vaibhav Kurhe" <vaibspidy@openmailbox.org>
"Vanita Rajpurohit" <vanita.rajpurohit9819@gmail.com>
"Bhavesh Bawadhane" <bbhavesh07@gmail.com>
"Sachin Patil" <sachpatil@openmailbox.org>
'Prajkta Patkar'<prajkta@riseup.net>
"""

from pyramid.config import Configurator
from gkwebapp.resources import get_root


def main(global_config, **settings):
    """ This function returns a WSGI application.

    It is usually called by the PasteDeploy framework during
    ``paster serve``.
    """
    settings = dict(settings)
    settings.setdefault('jinja2.i18n.domain', 'gkwebapp')

    config = Configurator(root_factory=get_root, settings=settings)
    config.add_translation_dirs('locale/')
    config.include('pyramid_jinja2')
    config.add_static_view('static', 'static')
    config.add_view('gkwebapp.views.views.my_view',
                    context='gkwebapp.resources.MyResource',
                    renderer="templates/mytemplate.jinja2")
    config.add_route('index', '/')
    config.add_route('locale', '/locale')
    config.add_route('theme', '/theme')
    config.add_route('addtheme', '/addtheme')
    config.add_route('backupfile', '/backupfile')
    config.add_route('recoveryfile', '/recoveryfile')
    config.add_route('invoice', '/invoice')
    config.add_route('existingorg', '/existingorg')
    config.add_route('showeditOrg', '/showeditOrg')
    config.add_route('showedituser', '/showedituser')
    config.add_route('getorgcode','/getorgcode')
    config.add_route('edituser','/edituser')
    config.add_route('editorganisation', '/editorganisation')
    config.add_route('orgexists', '/orgexists')
    config.add_route('oexists', '/oexists')
    config.add_route('findeditvoucher', '/findeditvoucher')
    config.add_route('editvoucher', '/editvoucher')
    config.add_route('createorg', '/createorg')
    config.add_route('yearcode', '/yearcode')
    config.add_route('login', '/login')
    config.add_route('catsearch', '/catsearch')
    config.add_route('product', '/product')
    config.add_route('updaterecon', '/updaterecon')
    config.add_route('orgdata', '/orgdata')
    config.add_route('closebooks', '/closebooks')
    config.add_route('deletevoucher', '/deletevoucher')
    config.add_route('showmainshell', '/showmainshell')
    config.add_route('showviewledger', '/showviewledger')
    config.add_route('stock', '/stock')
    config.add_route('printtrialbalancereport', '/printtrialbalancereport')
    config.add_route('cashflowreportprint', '/cashflowreportprint')
    config.add_route('listofaccountsprint', '/listofaccountsprint')
    config.add_route('printvouchers', '/printvouchers')
    config.add_route('showviewbankrecon', '/showviewbankrecon')
    config.add_route('showunclearedbankrecon', '/showunclearedbankrecon')
    config.add_route('viewdualledger', '/viewdualledger')
    config.add_route('showdualledgerreport', '/showdualledgerreport')
    config.add_route('showclearedbankrecon', '/showclearedbankrecon')
    config.add_route('showledgerreport', '/showledgerreport')
    config.add_route('showtrialbalance', '/showtrialbalance')
    config.add_route('showtrialbalancereport', '/showtrialbalancereport')
    config.add_route('showcashflow', '/showcashflow')
    config.add_route('showprofitloss', '/showprofitloss')
    config.add_route('showbalancesheet', '/showbalancesheet')
    config.add_route('showbalancesheetreport', '/showbalancesheetreport')
    config.add_route('showcashflowreport', '/showcashflowreport')
    config.add_route('showprofitlossreport', '/showprofitlossreport')
    config.add_route('showviewprojectstatement', '/showviewprojectstatement')
    config.add_route('showprojectstatementreport', '/showprojectstatementreport')
    config.add_route('showaccount', '/showaccount')
    config.add_route('showreport', '/showreport')
    config.add_route('shortcuts', '/shortcuts')
    config.add_route('authors', '/authors')
    config.add_route('license', '/license')
    config.add_route('about', '/about')
    config.add_route('deletecompany', '/deletecompany')
    config.add_route('showlistofaccounts', '/showlistofaccounts')
    config.add_route('showmultiacc', '/showmultiacc')
    config.add_route('multiacc', '/multiacc')
    config.add_route('accountexists', '/accountexists')
    config.add_route('accountpopup', '/accountpopup')
    config.add_route('showvoucher', '/showvoucher')
    config.add_route('showproject', '/showproject')
    config.add_route('viewproject', '/viewproject')
    config.add_route('editproject', '/editproject')
    config.add_route('addproject', '/addproject')
    config.add_route('delproject', '/delproject')
    config.add_route('lockvoucher', '/lockvoucher')
    config.add_route('viewvoucher', '/viewvoucher')
    config.add_route('addvoucher', '/addvoucher')
    config.add_route('addvoucherauto', '/addvoucherauto')
    config.add_route('showeditaccount', '/showeditaccount')
    config.add_route('getaccdetails', '/getaccdetails')
    config.add_route('getvouchers', '/getvouchers')
    config.add_route('getcjaccounts', '/getcjaccounts')
    config.add_route('showuser', '/showuser')
    config.add_route('createuser', '/createuser')
    config.add_route('addaccount', '/addaccount')
    config.add_route('category', '/category')
    config.add_route('deliverychallan', '/deliverychallan')
    config.add_route('editaccount', '/editaccount')
    config.add_route('deleteaccount', '/deleteaccount')
    config.add_route('getsubgroup','/getsubgroup')
    config.add_route('createorglogin','/createorglogin')
    config.add_route('userlogin','/userlogin')
    config.add_route('createadmin', '/createadmin')
    config.add_route('removeuser', '/removeuser')
    config.add_route('deleteuser', '/deleteuser')
    config.add_route('forgotpassword', '/forgotpassword')
    config.add_route('showclosebooks', '/showclosebooks')
    config.add_route('rollover', '/rollover')
    config.add_route('printlistofaccounts', '/printlistofaccounts')
    config.add_route('printsourcesandappfundreport', '/printsourcesandappfundreport')
    config.add_route('printconvbalsheetreport', '/printconvbalsheetreport')
    config.add_route('printmonthlyledgerreport', '/printmonthlyledgerreport')
    config.add_route('printprojectstatementreport', '/printprojectstatementreport')
    config.add_route('printprofitandloss', '/printprofitandloss')
    config.add_route('printledgerreport', '/printledgerreport')
    config.add_route('printtrialbalance', '/printtrialbalance')
    config.add_route('printcashflowreport', '/printcashflowreport')
    config.add_route('showdeletedvoucher', '/showdeletedvoucher')
    config.add_route('deleteorg', '/deleteorg')
    config.add_route('getattachment', '/getattachment')
    config.add_route('securityquestion', '/securityquestion')
    config.add_route('securityanswer', '/securityanswer')
    config.add_route('newpassword', '/newpassword')
    config.add_route('updateattachment', '/updateattachment')
    config.add_route('printprojectstatement', '/printprojectstatement')
    config.add_route('printledger', '/printledger')
    config.add_route('unitofmeasurements','/unitofmeasurements')
    config.add_route('godown', '/godown')
    config.add_route('budget', '/budget')
    config.add_route('customersuppliers','/customersuppliers')
    config.add_route('transfernotes','/transfernotes')
    config.add_route('cashmemos','/cashmemos')
    config.add_route('import','/import')
    config.add_route('exportledger','/exportledger')
    config.add_route('log','/log')
    config.add_route('del_unbilled','/del_unbilled')
    config.add_route('show_del_unbilled_report','/show_del_unbilled_report')
    config.add_route('show_del_cancelled_report','/show_del_cancelled_report')
    config.add_route('print_unbilled_deliveries_report','/print_unbilled_deliveries_report')
    config.add_route('print_cancelled_deliveries_report','/print_cancelled_deliveries_report')
    config.add_route('purchaseorder','/purchaseorder')
    config.add_route('salesorder','/salesorder')
    config.add_route('lastfivegodowns', '/lastfivegodowns')
    config.add_route('allorgcode','/allorgcode')
    config.add_route('listoforgselected','/listoforgselected')
    config.add_route('showconsolidationpopup','/showconsolidationpopup')
    config.add_route('rejectionnote','/rejectionnote')
    config.add_route('billwise','/billwise')
    config.add_route('gstsummary','/gstsummary')
    config.add_route('drcrnote','/drcrnote')
    config.add_route('genstats', '/genstats')
    config.add_route('viewgstreturns', '/viewgstreturns')
    config.add_route('gstreturns', '/gstreturns')
    config.add_route('dashboard', '/dashboard')
    config.add_route('export', '/export')
    config.scan('gkwebapp')
    return config.make_wsgi_app()
